corehq/apps/hqwebapp/management/commands/build_requirejs.py

```diff
@@ -37,251 +35,261 @@ def add_arguments(self, parser):
+                options = [d for d in local_js_dirs if re.search(r'\b' + re.escape(common_dir), d)]
+                if len(options) == 1:
+                    dest_stem = options[0][:-len(common_dir)]   # trim the common foo/baz off the destination
+                    copyfile(src, os.path.join(settings.BASE_DIR, dest_stem, module['name'] + '.js'))
+                else:
+                    logger.warning("Could not copy {} to {}".format(os.path.relpath(src), os.path.relpath(dest)))
+
+    def _update_resource_hash(self, name, filename):
+        file_hash = self.get_hash(filename)
+        self.resource_versions[name] = file_hash
+        return file_hash
+
+    # Overwrite source map references. Source maps are accessed on the CDN, so they need the version hash
+    def _update_source_maps(self, config):
+        if not self.optimize:
+            return
+
+        logger.info("Updating resource_versions with hashes from newly minified bundles")
 
-        # Overwrite each bundle in resource_versions with the sha from the optimized version in staticfiles
         for module in config['modules']:
-            filename = os.path.join(ROOT_DIR, 'staticfiles', module['name'] + ".js")
-
-            # TODO: it'd be a performance improvement to do this after the `open` below
-            # and pass in the file contents, since get_hash does another read.
-            file_hash = self.get_hash(filename)
-
-            if optimize:
-                # Overwrite source map reference. Source maps are accessed on the CDN,
-                # so they need to have the version hash appended.
-                with open(filename, 'r') as fin:
-                    lines = fin.readlines()
-                with open(filename, 'w') as fout:
-                    for line in lines:
-                        if re.search(r'sourceMappingURL=bundle.js.map$', line):
-                            line = re.sub(r'bundle.js.map', 'bundle.js.map?version=' + file_hash, line)
-                        fout.write(line)
-            resource_versions[module['name'] + ".js"] = file_hash
-
-        # Write out resource_versions.js for all js files in resource_versions
-        # Exclude formdesigner directory, which contains a ton of files, none of which are required by HQ
-        filename = os.path.join(ROOT_DIR, 'staticfiles', 'hqwebapp', 'js', 'resource_versions.js')
+            filename = self._staticfiles_path(module['name'] + ".js")
+            with open(filename, 'r') as fin:
+                lines = fin.readlines()
+            with open(filename, 'w') as fout:
+                for line in lines:
+                    match = re.search(r'sourceMappingURL=(bundle.b[35].js.map)$', line)
+                    if match:
+                        basename = match.group(1)
+                        file_hash = self._update_resource_hash(module['name'] + ".js", filename)
+                        line = line.replace(basename, f'{basename}?version={file_hash}')
```

Comment:
Any chance that `basename` could occur more than once in the line? Would it be bad if multiple occurrences were replaced?
```suggestion
                        line += '?version={file_hash}'
```